scalding-db-macros/README.md

```diff
@@ -0,0 +1,82 @@
+## Scalding JDBC Macros
+
+Provides macros to interop between Scala case classes and relational database / SQL column definitions.
+
+For a case class T, the macro-generated `ColumnDefinitionProvider[T]` provides:
+1. `ColumnDefinition`s for the corresponding DB table columns
+2. `ResultSetExtractor[T]` for extracting records from `java.sql.ResultSet` into objects of type `T`
+
+Also provided are `TupleConverter`, `TupleSetter` and `cascading.tuple.Fields` for use with Cascading.
+`DBTypeDescriptor[T]` is the top-level class that contains all of the above.
+
+### Illustration
+
+(in the REPL)
+
+Necessary imports:
+
+    scalding> import com.twitter.scalding.db_
+    scalding> import com.twitter.scalding.db.macros._
+
+Case class representing your DB schema:
+
+    scalding> case class ExampleDBRecord(
+       |   card_id: Long,
+       |   tweet_id: Long,
+       |   created_at: Option[java.util.Date],
+       |   deleted: Boolean = false)
+    defined class ExampleDBRecord
+
```

Comment:
what's the last step? How to I read or write from a Database? Seems like we are stopping short of clearly explaining the use of the whole package here.